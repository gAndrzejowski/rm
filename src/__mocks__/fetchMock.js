export default async uri => Promise.resolve({
  json: async () => Promise.resolve({ data: uri }),
});
