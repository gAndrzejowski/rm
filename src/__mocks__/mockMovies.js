const mockMovies = [
    {
        "id": 185567,
        "title": "Zulu",
        "tagline": "",
        "vote_average": 6.7,
        "vote_count": 200,
        "release_date": "2013-12-04",
        "poster_path": "https://image.tmdb.org/t/p/w500/xg7Dh7mjevDgznqw9JHYccNo9ZQ.jpg",
        "overview": "As a child, Ali Neuman narrowly escaped being murdered by Inkhata, a militant political party at war with Nelson Mandela's African National Congress. Only he and his mother survived the carnage of those years. But as with many survivors, the psychological scars remain.",
        "budget": 16000000,
        "revenue": 0,
        "genres": [
            "Crime",
            "Drama",
            "Thriller"
        ],
        "runtime": 110
    },
    {
        "id": 269149,
        "title": "Zootopia",
        "tagline": "Welcome to the urban jungle.",
        "vote_average": 7.7,
        "vote_count": 6795,
        "release_date": "2016-02-11",
        "poster_path": "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
        "overview": "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
        "budget": 150000000,
        "revenue": 1023784195,
        "genres": [
            "Animation",
            "Adventure",
            "Family",
            "Comedy"
        ],
        "runtime": 108
    },
    {
        "id": 19908,
        "title": "Zombieland",
        "tagline": "This place is so dead",
        "vote_average": 7.2,
        "vote_count": 4509,
        "release_date": "2009-10-07",
        "poster_path": "https://image.tmdb.org/t/p/w500/vUzzDpVrab1BOG3ogxhRGfLN94d.jpg",
        "overview": "Columbus has made a habit of running from what scares him. Tallahassee doesn't have fears. If he did, he'd kick their ever-living ass. In a world overrun by zombies, these two are perfectly evolved survivors. But now, they're about to stare down the most terrifying prospect of all: each other.",
        "budget": 23600000,
        "revenue": 102391382,
        "genres": [
            "Comedy",
            "Horror"
        ],
        "runtime": 88
    },
    {
        "id": 1949,
        "title": "Zodiac",
        "tagline": "There's more than one way to lose your life to a killer.",
        "vote_average": 7.3,
        "vote_count": 2780,
        "release_date": "2007-03-02",
        "poster_path": "https://image.tmdb.org/t/p/w500/bgLyOROfFQI3FqYL7jQbiaV8lkN.jpg",
        "overview": "The true story of the investigation of the \"Zodiac Killer\", a serial killer who terrified the San Francisco Bay Area, taunting police with his ciphers and letters.  The case becomes an obsession for four men as their lives and careers are built and destroyed by the endless trail of clues.",
        "budget": 65000000,
        "revenue": 84785914,
        "genres": [
            "Crime",
            "Drama",
            "Mystery",
            "Thriller"
        ],
        "runtime": 157
    },
    {
        "id": 97630,
        "title": "Zero Dark Thirty",
        "tagline": "The Greatest Manhunt in History",
        "vote_average": 6.7,
        "vote_count": 2028,
        "release_date": "2012-12-19",
        "poster_path": "https://image.tmdb.org/t/p/w500/yg6IDNucLAEj7E5loTyTnUW2pgb.jpg",
        "overview": "A chronicle of the decade-long hunt for al-Qaeda terrorist leader Osama bin Laden after the September 2001 attacks, and his death at the hands of the Navy S.E.A.L. Team 6 in May, 2011.",
        "budget": 40000000,
        "revenue": 132820716,
        "genres": [
            "Thriller",
            "Drama",
            "History"
        ],
        "runtime": 157
    },
    {
        "id": 6795,
        "title": "Zathura: A Space Adventure",
        "tagline": "Adventure Is Waiting",
        "vote_average": 6.1,
        "vote_count": 1080,
        "release_date": "2005-11-06",
        "poster_path": "https://image.tmdb.org/t/p/w500/amqgIuISRBt8tsZM6cTT6gO9WLR.jpg",
        "overview": "After their father is called into work, two young boys, Walter and Danny, are left in the care of their teenage sister, Lisa, and told they must stay inside. Walter and Danny, who anticipate a boring day, are shocked when they begin playing Zathura, a space-themed board game, which they realize has mystical powers when their house is shot into space. With the help of an astronaut, the boys attempt to return home.",
        "budget": 65000000,
        "revenue": 64321501,
        "genres": [
            "Family",
            "Fantasy",
            "Science Fiction",
            "Adventure"
        ],
        "runtime": 101
    },
    {
        "id": 133931,
        "title": "Zambezia",
        "tagline": "Airborne to be wild.",
        "vote_average": 5.2,
        "vote_count": 89,
        "release_date": "2012-06-05",
        "poster_path": "https://image.tmdb.org/t/p/w500/dLZTSTe4czdJ3Fr6vwy8XwVzLF5.jpg",
        "overview": "Set in a bustling bird city on the edge of the majestic Victoria Falls, \"Zambezia\" is the story of Kai - a na√Øve, but high-spirited young falcon who travels to the bird city of \"Zambezia\" where he discovers the truth about his origins and, in defending the city, learns how to be part of a community",
        "budget": 20000000,
        "revenue": 0,
        "genres": [
            "Comedy",
            "Animation",
            "Adventure",
            "Family"
        ],
        "runtime": 83
    },
    {
        "id": 10358,
        "title": "Zack and Miri Make a Porno",
        "tagline": "What would you do to get out of debt?",
        "vote_average": 6.1,
        "vote_count": 894,
        "release_date": "2008-09-18",
        "poster_path": "https://image.tmdb.org/t/p/w500/zxwEJBhSSu5AuKTTd4ugWaCqmg.jpg",
        "overview": "Lifelong platonic friends Zack and Miri look to solve their respective cash-flow problems by making an adult film together. As the cameras roll, however, the duo begin to sense that they may have more feelings for each other than they previously thought.",
        "budget": 24000000,
        "revenue": 42105111,
        "genres": [
            "Comedy"
        ],
        "runtime": 102
    }
]
export default mockMovies;