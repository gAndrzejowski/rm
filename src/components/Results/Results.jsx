import React from 'react';
import PropTypes from 'prop-types';
import NoMovies from '../NoMovies/NoMovies';
import Movie from '../Movie/Movie'
import styles from './Results.scss';

const Results = ({results, chooseMovie}) => (
        <article className={styles.list}>
            {results.length===0 ? <NoMovies /> : results.map( (e, i) => (
                <Movie
                    key={e.id}
                    chooseMovie={() => chooseMovie(i)}
                    data={e}
                />))
            }

        </article>
    );

Results.propTypes = {
    results: PropTypes.array,
    chooseMovie: PropTypes.func.isRequired
};

export default Results;