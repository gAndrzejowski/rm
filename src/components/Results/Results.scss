.list {
  display:flex;
  flex-flow:row wrap;
  justify-content: space-between;
  align-items: flex-start;
  background:white;
  padding:2em 5vw;
  min-height:50vh;
}