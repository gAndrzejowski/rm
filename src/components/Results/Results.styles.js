const styles = {
  list: {
    display: 'flex',
    flexFlow: 'row wrap',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    background: 'white',
    padding: ['2em', '5vw'],
    minHeight: '50vh',
  },
};

export default styles;
