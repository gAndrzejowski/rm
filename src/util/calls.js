export const swagger = async settings => {
    const {
        sortBy,
        sortOrder,
        search,
        searchBy,
        filter,
    } = settings;
    const response = await fetch(`http://react-cdp-api.herokuapp.com/movies?search=${search || ''}&searchBy=${searchBy || 'title'}${filter ? ('&filter=' + filter) : ''}&sortBy=${sortBy || 'title'}&sortOrder=${sortOrder || 'desc'}`);
    return await response.json();
};

export default swagger;