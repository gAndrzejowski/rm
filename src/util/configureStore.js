import { createStore, applyMiddleware } from 'redux';
import persistStore from './middleware';
import rootReducer from '../reducers/rootReducer';
import { hydrateStore, applyPreloaded } from '../actions/creators';

export default (preloadedState :Object<any>) :Object<any> => {
  const store = createStore(rootReducer, applyMiddleware(persistStore));

  store.dispatch(hydrateStore());
  store.dispatch(applyPreloaded(preloadedState));
  return store;
};
