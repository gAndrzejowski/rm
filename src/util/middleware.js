const persistStore = store => next => (action) => {
  const result = next(action);
  window.localStorage.setItem('netflixRouletteStore', JSON.stringify(store.getState()));
  return result;
};
export default persistStore;
