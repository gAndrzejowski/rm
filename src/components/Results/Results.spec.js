import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router';
import mockMovies from '../../__mocks__/mockMovies';
import { Results, mapStateToProps, mapDispatchToProps } from './Results';
import fetchMock from '../../__mocks__/fetchMock';
import {
  getMovies,
  setCurrentMovie,
  setSearchCriterion,
  setSearchTxt,
} from '../../actions/creators';
import { swaggerBase } from '../../util/calls';

describe('Results component', () => {
  it('renders correctly when there are movies to display', () => {
    const tree = renderer.create(
      <Router>
        <Results
          chooseMovie={jest.fn()}
          results={mockMovies}
          getSameGenre={jest.fn()}
          setSearchTxt={jest.fn()}
          setSearchCriterion={jest.fn()}
        />
      </Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when there are no movies to display', () => {
    const tree = renderer.create(
      <Router>
        <Results
          chooseMovie={jest.fn()}
          results={[]}
          getSameGenre={jest.fn()}
          setTxt={jest.fn()}
          setCrit={jest.fn()}
        />
      </Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls chooseMovie, getSameGenre, setSearchTxt and setSearchCriterion props with the right data after calling one of the child components chooseMovie props', () => {
    const wrapper = shallow(
      <Router>
        <Results
          chooseMovie={jest.fn()}
          results={mockMovies}
          getSameGenre={jest.fn()}
          setTxt={jest.fn()}
          setCrit={jest.fn()}
        />
      </Router>);
    const results = wrapper.find('Results').dive();

    results.find('Movie').at(3).prop('chooseMovie')();
    results.find('Movie').at(6).prop('chooseMovie')();

    expect(wrapper.find('Results').prop('chooseMovie').mock.calls[0][0]).toEqual(mockMovies[3]);
    expect(wrapper.find('Results').prop('chooseMovie').mock.calls[1][0]).toEqual(mockMovies[6]);

    expect(wrapper.find('Results').prop('getSameGenre').mock.calls[0][0]).toBe(mockMovies[3].genres[0]);
    expect(wrapper.find('Results').prop('getSameGenre').mock.calls[1][0]).toBe(mockMovies[6].genres[0]);

    expect(wrapper.find('Results').prop('setTxt').mock.calls[0][0]).toBe(mockMovies[3].genres[0]);
    expect(wrapper.find('Results').prop('setTxt').mock.calls[1][0]).toBe(mockMovies[6].genres[0]);

    expect(wrapper.find('Results').prop('setCrit').mock.calls.length).toBe(2);
  });
});

describe('mapStateToProps function for Results', () => {
  it('takes results array from movies store', () => {
    const fakeState = {
      movies: {
        results: [1, 3, 5, 15],
      },
    };
    expect(mapStateToProps(fakeState)).toEqual({
      results: [1, 3, 5, 15],
    });
  });
});

describe('mapDispatchToProps for Results', () => {
  beforeAll(() => {
    global.fetch = jest.fn();
    global.fetch.mockImplementation(fetchMock);
  });
  it('creates props object with chooseMovie, getSameGenre, setSearchTxt and setSearchCriterion methods', () => {
    const props = mapDispatchToProps();
    expect(props).toEqual({
      chooseMovie: expect.any(Function),
      getSameGenre: expect.any(Function),
      setTxt: expect.any(Function),
      setCrit: expect.any(Function),
    });
  });
  it('creates a chooseMovie function which dispatches a result of setCurrentMovie action creator with given data', () => {
    const fakeDispatch = jest.fn();
    const props = mapDispatchToProps(fakeDispatch);
    props.chooseMovie('totally a movie');
    expect(fakeDispatch).toHaveBeenCalledTimes(1);
    expect(fakeDispatch.mock.calls[0][0]).toEqual(setCurrentMovie('totally a movie'));
  });
  it('creates a setSearchTxt function which dispatches a result of setSearchTxt action creator with given string', () => {
    const fakeDispatch = jest.fn();
    const props = mapDispatchToProps(fakeDispatch);
    props.setTxt('Dracula related movies');
    expect(fakeDispatch).toHaveBeenCalledTimes(1);
    expect(fakeDispatch.mock.calls[0][0]).toEqual(setSearchTxt('Dracula related movies'));
  });
  it('creates a setSearchCriterion function wchich dispatches a result of setSearchCriterion action creator with \'genres\' param', () => {
    const fakeDispatch = jest.fn();
    const props = mapDispatchToProps(fakeDispatch);
    props.setCrit();
    expect(fakeDispatch).toHaveBeenCalledTimes(1);
    expect(fakeDispatch.mock.calls[0][0]).toEqual(setSearchCriterion('genres'));
  });
  it('creates a getSameGenre function which calls the swagger api for movies of same genre, and dispatches getMoviesies with results', async () => {
    const fakeDispatch = jest.fn();
    const props = mapDispatchToProps(fakeDispatch);
    await props.getSameGenre('Action');
    expect(fakeDispatch).toHaveBeenCalledTimes(1);
    const uri = `${swaggerBase}/movies?search=Action&searchBy=genres&sortBy=title&sortOrder=desc`;
    expect(fakeDispatch.mock.calls[0][0]).toEqual(getMovies(uri));
  });
});
