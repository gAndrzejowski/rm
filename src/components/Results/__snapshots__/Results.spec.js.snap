// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders correctly when there are movies to display 1`] = `
<article
  className="list"
>
  <a
    className="panel"
    href="/film/185567"
    onClick={[Function]}
  >
    <img
      alt="Zulu"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/xg7Dh7mjevDgznqw9JHYccNo9ZQ.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zulu
        </div>
        <div
          className="releaseDate"
        >
          2013
        </div>
      </div>
      <p
        className="genres"
      >
        Crime, Drama, Thriller
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/269149"
    onClick={[Function]}
  >
    <img
      alt="Zootopia"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zootopia
        </div>
        <div
          className="releaseDate"
        >
          2016
        </div>
      </div>
      <p
        className="genres"
      >
        Animation, Adventure, Family, Comedy
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/19908"
    onClick={[Function]}
  >
    <img
      alt="Zombieland"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/vUzzDpVrab1BOG3ogxhRGfLN94d.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zombieland
        </div>
        <div
          className="releaseDate"
        >
          2009
        </div>
      </div>
      <p
        className="genres"
      >
        Comedy, Horror
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/1949"
    onClick={[Function]}
  >
    <img
      alt="Zodiac"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/bgLyOROfFQI3FqYL7jQbiaV8lkN.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zodiac
        </div>
        <div
          className="releaseDate"
        >
          2007
        </div>
      </div>
      <p
        className="genres"
      >
        Crime, Drama, Mystery, Thriller
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/97630"
    onClick={[Function]}
  >
    <img
      alt="Zero Dark Thirty"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/yg6IDNucLAEj7E5loTyTnUW2pgb.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zero Dark Thirty
        </div>
        <div
          className="releaseDate"
        >
          2012
        </div>
      </div>
      <p
        className="genres"
      >
        Thriller, Drama, History
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/6795"
    onClick={[Function]}
  >
    <img
      alt="Zathura: A Space Adventure"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/amqgIuISRBt8tsZM6cTT6gO9WLR.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zathura: A Space Adventure
        </div>
        <div
          className="releaseDate"
        >
          2005
        </div>
      </div>
      <p
        className="genres"
      >
        Family, Fantasy, Science Fiction, Adventure
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/133931"
    onClick={[Function]}
  >
    <img
      alt="Zambezia"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/dLZTSTe4czdJ3Fr6vwy8XwVzLF5.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zambezia
        </div>
        <div
          className="releaseDate"
        >
          2012
        </div>
      </div>
      <p
        className="genres"
      >
        Comedy, Animation, Adventure, Family
      </p>
    </div>
  </a>
  <a
    className="panel"
    href="/film/10358"
    onClick={[Function]}
  >
    <img
      alt="Zack and Miri Make a Porno"
      className="poster"
      src="https://image.tmdb.org/t/p/w500/zxwEJBhSSu5AuKTTd4ugWaCqmg.jpg"
    />
    <div>
      <div
        className="info"
      >
        <div
          className="title"
        >
          Zack and Miri Make a Porno
        </div>
        <div
          className="releaseDate"
        >
          2008
        </div>
      </div>
      <p
        className="genres"
      >
        Comedy
      </p>
    </div>
  </a>
</article>
`;

exports[`Results component renders correctly when there are no movies to display 1`] = `
<article
  className="list"
>
  <article
    className="noResults"
  >
    <h2>
      No films found
    </h2>
  </article>
</article>
`;
